allennlp/run.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, Union
```

Comment:
@joelgrus our goal isn't specifically to make allennlp pip-installable, but rather to make it easy for users to get started with allennlp.  If pip is the easiest way to get them going, then we should go with it!  You probably know best about what's pythonic.